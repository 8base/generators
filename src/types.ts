import { TableSchema } from '@8base/utils';

export interface GeneratorsCommonConfig {
  deep?: number;
  withMeta?: boolean;
}

export type GeneratorsConfig = {
  includeColumns?: string[],
} & GeneratorsCommonConfig;

export interface GeneratorsData {
  tablesList: TableSchema[];
  tableName: string;
  screenName?: string;
}

export interface ScreenTable {
  id?: string;
  tableName: string;
  screenName: string;
  routeUrl?: string;
  tableFields?: string[];
  formFields?: string[];
}
