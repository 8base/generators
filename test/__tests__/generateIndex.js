

import { generateIndex } from '../../src';
import { tablesSchema } from '../__fixtures__/tablesSchema';


it('should generate table by the table name', () => {
  const generatedTable = generateIndex({
    tablesList: tablesSchema,
    tableName: 'Properties',
    screenName: 'Properties',
  });

  expect(generatedTable).toMatchSnapshot();
});

