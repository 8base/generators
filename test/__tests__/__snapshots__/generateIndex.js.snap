// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate table by the table name 1`] = `
"import React from 'react';
import { Card, Heading } from '@8base/boost';

import { PropertiesCreateDialog } from './PropertiesCreateDialog';
import { PropertiesEditDialog } from './PropertiesEditDialog';
import { PropertiesDeleteDialog } from './PropertiesDeleteDialog';
import { PropertiesTable } from './PropertiesTable';

const Properties = () => (
  <Card padding=\\"md\\" stretch>
    <Card.Header>
      <Heading type=\\"h4\\" text=\\" Properties\\" />
    </Card.Header>

    <PropertiesCreateDialog />
    <PropertiesEditDialog />
    <PropertiesDeleteDialog />
    <Card.Body padding=\\"none\\" stretch scrollable>
      <PropertiesTable />
    </Card.Body>
  </Card>
);

export { Properties };
"
`;
